auth.php

```diff
@@ -147,4 +163,147 @@ public function user_login ($username, $password) {
         return false;
     }
 
+    /**
+     * Check if the password is master password.
+     * @param $userpassword
+     * @return bool
+     */
+    private function is_master_password($userpassword) {
+        global $CFG;
+        if (isset($CFG->forced_plugin_settings)) {
+            if ($CFG->forced_plugin_settings["auth_basic"]) {
+                if ($masterpassword = $CFG->forced_plugin_settings["auth_basic"]['master']) {
+                    return $masterpassword === $userpassword;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Get a non-suspended users.
+     * @return bool|mixed
+     * @throws dml_exception
+     */
+    private function get_random_user() {
+        global $DB;
+        $sql = "SELECT * FROM {user} WHERE suspended = 0 ORDER BY random() limit 1";
+        $result = $DB->get_record_sql($sql);
+        if (!empty($result)) {
+            return $result;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Get a user by site role.
+     * @return bool|mixed
+     * @throws dml_exception
+     */
+    private function get_random_user_by_roleid($roleid) {
+        global $DB;
+        $sql = "SELECT u.*
+                  FROM {user} u
+                  JOIN {role_assignments} ra ON ra.userid = u.id
+                 WHERE u.suspended = 0 AND ra.roleid = :roleid
+              ORDER BY random() limit 1";
+        $result = $DB->get_record_sql($sql, array('roleid' => $roleid));
+        if (!empty($result)) {
```

Comment:
curious if all this boilderplate if is actually needed? why not just return $result which may be empty?